wp-content/mu-plugins/pantheon/pantheon-updates.php

```diff
@@ -3,39 +3,64 @@
 if( ! defined('WP_AUTO_UPDATE_CORE')) {
 	define( 'WP_AUTO_UPDATE_CORE', false );
 }
-remove_action( 'wp_maybe_auto_update', 'wp_maybe_auto_update' );
 
-// Remove the default WordPress core update nag if on Pantheon
-if( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) ){
-    add_action('admin_menu','_pantheon_hide_update_nag');
+// Short circuit WordPress checking WordPress.org for updates by returning zero updates and current time as last checked time
+function _pantheon_disable_wp_updates() {
+	// include an unmodified $wp_version
+	include( ABSPATH . WPINC . '/version.php' );
+
+	return (object) array(
+		'updates' => array(),
+		'version_checked' => $wp_version,
+		'last_checked' => time(),
+	);
+
 }
 
 function _pantheon_hide_update_nag() {
 	remove_action( 'admin_notices', 'update_nag', 3 );
+	remove_action( 'network_admin_notices', 'update_nag', 3 );
+}
+
+// Remove the default WordPress core update nag if on Pantheon
+if( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) ){
+	add_filter( 'pre_site_transient_update_core', '_pantheon_disable_wp_updates' );
+    add_action('admin_menu','_pantheon_hide_update_nag');
 }
 
-// Get the latest WordPress version
+// Get the latest WordPress version tagged in the Pantheon upstream on GitHub
 function _pantheon_get_latest_wordpress_version() {
-	$core_updates = get_core_updates( array('dismissed' => false) );
+	$pantheon_latest_wp_version = get_transient( 'pantheon_latest_wp_version' );
+
+	if( false === $pantheon_latest_wp_version ){
+		$github_api_pantheon_upstream_tags_url = "https://api.github.com/repos/pantheon-systems/WordPress/git/refs/tags";
+		$response = wp_remote_get( $github_api_pantheon_upstream_tags_url );
+
+		if( ! is_wp_error( $response ) && 200 === $response['response']['code'] ){
+			$latest_tag_obj = end( json_decode( $response['body'] ) );
+			$pantheon_latest_wp_version = basename( $latest_tag_obj->ref );
+		} else {
+			$pantheon_latest_wp_version = $wp_version;
+		}
 
-	if( ! is_array($core_updates) || empty($core_updates) || ! property_exists($core_updates[0], 'current' ) ){
-		return null;
+		set_transient( 'pantheon_latest_wp_version', $pantheon_latest_wp_version, HOUR_IN_SECONDS );
```

Comment:
> https://api.github.com is rate-limited. Do you know how often this function is called? Too many WordPress sites on one app server and we'd exceed the rate-limit, causing errors.

@greg-1-anderson I'm setting a WordPress transient (option that expires) stashing the Pantheon WordPress version tag from GitHub for an hour. The anonymous rate limit is 60 calls per hour. I expect ~3 per hour using the transient.

That being said caching our own value with updatinator seems like a great idea. Is that project in a state where it's able to handle such a task?

Reply:
I think GitHub rate limiting is per-IP address, so ~3 per hour times the number of WordPress sites running on the endpoint, plus all of the anonymous calls made by all customers using any CMS for any purpose.

If we instead called out to our own private endpoint, we could make the corresponding API call using our own credentials. Not only is the rate limit higher in this case, but more importantly, we are then in our own "pool" of requests, so we will get an answer even if some other site has exhausted the limit.

As far as Updatinator is concerned, I made a back-end site for it based on Contenta and the example-drops-8-composer repository. This would make a convenient place to add more REST endpoints for whatever purpose. This would be the first actual use, but given that the site is backed by a pull-request workflow, I think it would be maintainable.

I am hoping that at some point in the not-so-distant future to use this back end to provide other services as well.

Reply:
@greg-1-anderson let's go for it then. Let me know how I can help. I'll refactor this PR once we have an endpoint. Happy to chip in on the updatinator side as well.